vertical-pod-autoscaler/e2e/v1/actuation.go

```diff
@@ -278,6 +280,79 @@ var _ = ActuationSuiteE2eDescribe("Actuation", func() {
 		ginkgo.By(fmt.Sprintf("Waiting for pods to be evicted, hoping it won't happen, sleep for %s", VpaEvictionTimeout.String()))
 		CheckNoPodsEvicted(f, MakePodSet(podList))
 	})
+
+	ginkgo.It("does not act on injected sidecars", func() {
+		const (
+			// TODO(krzysied): Update the image url when the agnhost:2.10 image
+			// is promoted to the k8s-e2e-test-images repository.
+			agnhostImage  = "gcr.io/k8s-staging-e2e-test-images/agnhost:2.10"
+			sidecarParam  = "--sidecar-image=k8s.gcr.io/pause:3.1"
+			sidecarName   = "webhook-added-sidecar"
+			servicePort   = int32(8443)
+			containerPort = int32(8444)
+		)
+
+		ginkgo.By("Setting up Webhook for sidecar injection")
+
+		client := f.ClientSet
+		namespaceName := f.Namespace.Name
+		defer utils.CleanWebhookTest(client, namespaceName)
+
+		// Make sure the namespace created for the test is labeled to be selected by the webhooks.
+		utils.LabelNamespace(f, f.Namespace.Name)
+		utils.CreateWebhookConfigurationReadyNamespace(f)
+
+		ginkgo.By("Setting up server cert")
+		context := utils.SetupWebhookCert(namespaceName)
+		utils.CreateAuthReaderRoleBinding(f, namespaceName)
+
+		utils.DeployWebhookAndService(f, agnhostImage, context, servicePort, containerPort, sidecarParam)
+
+		// Webhook must be placed after vpa webhook. Webhooks are registered alphabetically.
+		// Use name that starts with "z".
+		webhookCleanup := utils.RegisterMutatingWebhookForPod(f, "z-sidecar-injection-webhook", context, servicePort)
+		defer webhookCleanup()
+
+		ginkgo.By("Setting up a hamster deployment")
+
+		SetupVPA(f, "100m", vpa_types.UpdateModeAuto, hamsterTargetRef)
+		SetupHamsterDeployment(f, "100m", "100Mi", defaultHamsterReplicas)
+
+		podList, err := GetHamsterPods(f)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		for _, pod := range podList.Items {
+			observedContainers, ok := pod.GetAnnotations()[annotations.VpaObservedContainersLabel]
+			gomega.Expect(ok).To(gomega.Equal(true))
+			containers, err := annotations.ParseVpaObservedContainersValue(observedContainers)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(containers).To(gomega.HaveLen(1))
+			gomega.Expect(pod.Spec.Containers).To(gomega.HaveLen(2))
+		}
+
+		ginkgo.By("Adding recommendation for " + sidecarName)
+
+		vpa, err := getVpaClientSet(f).AutoscalingV1().VerticalPodAutoscalers(f.Namespace.Name).
```

Comment:
Why not set recommendation up in the first place?

Reply:
I though it will be less code with creat + patch. It turned out to be not the greatest idea...
I moved the sidecar recommendation to the hamster vpa setup.